internal/auth0/user/resource_permissions.go

```diff
@@ -0,0 +1,174 @@
+package user
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/auth0/go-auth0"
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+	"github.com/auth0/terraform-provider-auth0/internal/value"
+)
+
+// NewPermissionsResource will return a new auth0_connection_client resource.
+func NewPermissionsResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of the user to associate the permission to.",
+			},
+			"permissions": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				ForceNew:    true,
+				Description: "List of API permissions granted to the user.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Name of permission.",
+						},
+						"resource_server_identifier": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Resource server identifier associated with the permission.",
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Description of the permission.",
+						},
+						"resource_server_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Name of resource server that the permission is associated with.",
+						},
+					},
+				},
+			},
+		},
+		CreateContext: createUserPermissions,
+		ReadContext:   readUserPermissions,
+		DeleteContext: deleteUserPermissions,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: "With this resource, you can manage all of a user's permissions.",
+	}
+}
+
+func createUserPermissions(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
+
+	userID := data.Get("user_id").(string)
+
+	if !data.HasChange("permissions") {
+		return nil
+	}
```

Comment:
This will never be the case when creating user permissions, the permissions will always have a change.
```suggestion
```